sys/include/net/gnrc/ipv6/nib/ft.h

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_ipv6_nib_ft    Forwarding table
+ * @ingroup     net_gnrc_ipv6_nib
+ * @brief
+ * @{
+ *
+ * @file
+ * @brief   Forwarding table definitions
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef NET_GNRC_IPV6_NIB_FT_H
+#define NET_GNRC_IPV6_NIB_FT_H
+
+#include <stdint.h>
+
+#include "net/gnrc/pkt.h"
+#include "net/ipv6/addr.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Forwarding table entry view on NIB
+ */
+typedef struct {
+    ipv6_addr_t dst;        /**< destination or prefix */
+    ipv6_addr_t next_hop;   /**< next hop to gnrc_ipv6_nib_ft_t::dst */
+    uint8_t dst_len;        /**< prefix-length in bits of
+                             *   gnrc_ipv6_nib_ft_t::dst */
+    uint8_t primary;        /**< != 0 if gnrc_ipv6_nib_ft_t::dst is preferred
+                             *   default route */
+    uint16_t iface;         /**< interface to gnrc_ipv6_nib_ft_t::next_hop */
+} gnrc_ipv6_nib_ft_t;
+
+/**
+ * @brief   Gets the best matching forwarding table entry to a destination
+ *
+ * @pre `(dst != NULL) && (fte != NULL)`
+ *
+ * @param[in] dst   The destination.
+ * @param[in] ctx   Packet that is supposed to go to that destination
+ *                  (is handed over to a reactive routing protocol if one exist
```

Comment:
> Why isn't this named pkt, but `ctx`, if the documentation says packet?

Because it is the context of the forwarding step, but I don't care. I can call it pkt if you want.

> Can this be const probably?

Maybe, but I'm not sure if this is guaranteed for every reactive routing protocol (even RPL adds stuff like routing headers to a packet and that's not even reactive).

Reply:
> Because it is the context of the forwarding step, but I don't care. I can call it pkt if you want.

Yes, please.